### PR TITLE
Run `apt update` only once

### PR DESCRIPTION
**Pull request by jadarsie**
Commits: 
- Formatted script using "Bash beautify" extension
- Added function "retrycmd_if_failure_exit_code" for future use
- Execute "apt update" just once, download packages in parallel

### CODE REVIEW

File: kubernetes/template/DeploymentTemplates/script.sh [hunk 1]
```diff
@@ -4,21 +4,21 @@ set -e
 function collect_deployment_and_operations {
     DEPLOYLOGSDIR=/var/log/azure/arm-deployments/
     sudo mkdir -p /var/log/azure/arm-deployments/
-
+
```
Comment (honcao) [inline kubernetes/template/DeploymentTemplates/script.sh]: could you please use update the customData in template?

File: kubernetes/template/DeploymentTemplates/script.sh [hunk 2]
```diff
@@ -124,130 +150,114 @@ convert_to_cert() {
...
-retrycmd_if_failure 5 10 sudo apt-get update 
+log_level -i "Update package information from sources."
+retrycmd_if_failure 5 10 sudo apt-get update
 
-log_level -i "Installing azure cli...."
-retrycmd_if_failure 5 10 sudo apt-get install azure-cli
+#####################################################################################
+# Install all prerequisites
+log_level -i "Installing azure-cli and its dependencies."
+retrycmd_if_failure 5 10 sudo apt-get install -y \
```
Comment (deaborch) [inline kubernetes/template/DeploymentTemplates/script.sh]: @jadarsie Quick question, what is the retry logic here when one of the packages fail to install. Does is count as a failed package install or does it move on to the next package and installs it.

File: kubernetes/template/DeploymentTemplates/script.sh (hunk 1 above)
Comment (jadarsie) [reply kubernetes/template/DeploymentTemplates/script.sh]: I will have a lot more changes for this file in the next few days. 
Probably fine as we are not releasing anytime soon?
I am hoping to automate the process

File: kubernetes/template/DeploymentTemplates/script.sh [hunk 3]
```diff
@@ -124,130 +150,114 @@ convert_to_cert() {
...
 sudo tee /etc/apt/sources.list.d/azure-cli.list
 
-#Get Microsoft signing key
+# Download Microsoft signing key
 log_level -i "Get the Microsoft signing key."
+RECV_KEY=BC528686B50D79E339D3721CEB3E94ADBE1229CF
 retrycmd_if_failure 5 10 sudo apt-key --keyring /etc/apt/trusted.gpg.d/Microsoft.gpg adv --keyserver packages.microsoft.com --recv-keys $RECV_KEY
 
-log_level -i "Update system again to latest."
-retrycmd_if_failure 5 10 sudo apt-get update
```
Comment (deaborch) [inline kubernetes/template/DeploymentTemplates/script.sh]: Also, i find doing apt-get update after package installations tend to fix any missing packages and version issues after the previous packages have been installed. If we are removing it, what guarantee do we have that the packages install have not put the system in a bad state?? Just a general comment

File: kubernetes/template/DeploymentTemplates/script.sh (hunk 1 above)
Comment (honcao) [reply kubernetes/template/DeploymentTemplates/script.sh]: Ok.

File: kubernetes/template/DeploymentTemplates/script.sh (hunk 2 above)
Comment (jadarsie) [reply kubernetes/template/DeploymentTemplates/script.sh]: Not sure where my answer went. I'll type it again.

`retrycmd_if_failure` retries the entire thing
`apt-get install` aborts the download process when it finds the first missing package
I do not know what `apt-get install` does if the installation process fails

File: kubernetes/template/DeploymentTemplates/script.sh (hunk 3 above)
Comment (jadarsie) [reply kubernetes/template/DeploymentTemplates/script.sh]: I do not think `apt update` does that.
Quick link: https://linux.die.net/man/8/apt-get#update

Please share with me any resource supporting that claim. I'll like to learn

**Comment by jadarsie** [comment]
Closing. There is a new larger PR that includes these changes.

**Pull request closed without merging**